pep-0307.txt

```diff
@@ -5,767 +5,808 @@ Last-Modified: $Date$
-        trivial classic class (one without any methods or instance
-        variables) and then use __class__ assignment to change its
-        class to the desired class.  This creates an instance of the
-        desired class with an empty __dict__ whose __init__ has not
-        been called.
-
-      - However, if the class has a method named __getinitargs__, the
-        above trick is not used, and a class instance is created by
-        using the tuple returned by __getinitargs__ as an argument
-        list to the class constructor.  This is done even if
-        __getinitargs__ returns an empty tuple -- a __getinitargs__
-        method that returns () is not equivalent to not having
-        __getinitargs__ at all.  __getinitargs__ *must* return a
-        tuple.
-
-      - In restricted execution mode, the trick from the first bullet
-        doesn't work; in this case, the class constructor is called
-        with an empty argument list if no __getinitargs__ method
-        exists.  This means that in order for a classic class to be
-        unpicklable in restricted execution mode, it must either
-        implement __getinitargs__ or its constructor (i.e., its
-        __init__ method) must be callable without arguments.
+----------------------------------------
+
+This case is the same for all protocols, and is unchanged from
+Python 2.1.
+
+For classic classes, ``__reduce__`` is not used.  Instead, classic
+classes can customize their pickling by providing methods named
+``__getstate__``, ``__setstate__`` and ``__getinitargs__``.  Absent these, a
+default pickling strategy for classic class instances is
+implemented that works as long as all instance variables are
+picklable.  This default strategy is documented in terms of
+default implementations of ``__getstate__`` and ``__setstate__``.
+
+The primary ways to customize pickling of classic class instances
+is by specifying ``__getstate__`` and/or ``__setstate__`` methods.  It is
+fine if a class implements one of these but not the other, as long
+as it is compatible with the default version.
+
+The ``__getstate__`` method
+'''''''''''''''''''''''''''
+
+The ``__getstate__`` method should return a picklable value
+representing the object's state without referencing the object
+itself.  If no ``__getstate__`` method exists, a default
+implementation is used that returns ``self.__dict__``.
+
+The ``__setstate__`` method
+'''''''''''''''''''''''''''
+
+The ``__setstate__`` method should take one argument; it will be
+called with the value returned by ``__getstate__`` (or its default
+implementation).
+
+If no ``__setstate__`` method exists, a default implementation is
+provided that assumes the state is a dictionary mapping instance
+variable names to values.  The default implementation tries two
+things:
+
+- First, it tries to call ``self.__dict__.update(state)``.
+
+- If the update() call fails with a RuntimeError exception, it
```

Comment:
You should probably wrap update(), RuntimeError, setattr(...) and (key, value) with backticks in order to be consistent with the rest of the file.